modules/OCPP/OCPP.cpp

```diff
@@ -624,15 +625,19 @@ void OCPP::ready() {
 
     this->charge_point->register_disable_evse_callback([this](int32_t connector) {
```

Comment:
I think this callback might be used by both the OCPP `ChangeAvailability.req` and firmware update? If so it might be necessary to extend the callback to cater for source.

Reply:
Charge Availability, Update Firmware, and Signed Update Firmware are all requests from the CSMS. i.e. the same source. (note firmware update only disables availability - enabled following restart of EVerest)

Reply:
I'd thought it would be necessary for the following two use cases:

- If a firmware upgrade is ongoing (the charge point is unavailable), I would not expect the CSMS to be able to make the charge point available. 
- Similarly, if the CSMS has made the charge point unavailable, I would not expect a failed firmware upgrade to make it available. 

I don't think they're specifically covered in OCPP 1.6 (unless one infers from pg42 that the state of internal unavailability sources is distinct).

Reply:
When a FW upgrade is ongoing libocpp could/should reject change availability requests. That wouldn't require a change the disable EVSE callback.
Failure of a firmware upgrade doesn't cause the charge point to be come available (from libocpp). It is only made unavailable once the update has been successfully downloaded and verified.